LostCities/Game.py

```diff
@@ -0,0 +1,351 @@
+
+    def discard_card(self, other: Discardpile, player: Player) -> None:
+        """
+        The function used to discard a specific player's card
+
+        Args:
+            other (Discardpile): the chosen discardpile that the player wants to discard their card onto
+            player (Player): The current player
+        """
+        print("here are the discard piles:")
+        for i, discard_pile in enumerate(other):
+            if len(discard_pile.cardarray) == 0:
+                print(f"{i+1} {discard_pile.color} Pile üres")
+            else:
+                for i, card in enumerate(discard_pile.cardarray):
+                    print(f"{i} card is: {card}")
+
+        while True:
+            which_discardcolumn = input(
+                "Pick a discardpile from 1 to 5, if you see this again, it means you picked a discardpile, with a color that you don't have, pick another one"
+            )
+            color_to_check = other[int(which_discardcolumn) - 1].color
+            has_color = any(card.color == color_to_check for card in player.cardarray)
+            if has_color:
+                break
+
+        print("your cards: ")
+        for i in range(len(player.cardarray)):
+            print(f"{i +1}. card: {player.cardarray[i]}")
+
+        choice_str = input("choose which one you want to discard: ")
+        choice = int(choice_str)
+
+        while (
+            other[int(which_discardcolumn) - 1].color
+            != player.cardarray[choice - 1].color
+        ):
+            choice_str = input(
+                "Card's color is not equal to the chosen card's color, pick again: "
+            )
+            choice = int(choice_str)
+
+        if choice <= 0 or choice > len(player.cardarray):
+            print("bad choice, choose again")
+            choice_str = input()
+            choice = int(choice_str)
+        else:
+            other[int(which_discardcolumn) - 1].cardarray.append(
+                (player.cardarray[choice - 1])
+            )
+            player.cardarray.pop(choice - 1)
+
+    def play_card(self, other: Playercardpile, player: Player) -> None:
+        """
+            Plays a card from the player's card pile onto another player's card pile.
+
+            Args:
+                other (Playercardpile): The other player's card pile where the card will be played.
+                player (Player): The player who will play the card.
+
+            Returns:
+        None
+        """
+        print("here are the player columns:")
+        for i, player_pile in enumerate(other):
+            if len(player_pile.cardarray) == 0:
+                print(f"{i+1} {player_pile.color} Pile üres")
+            else:
+                print(player_pile.cardarray)
+
+        while True:
+            which_playercolumn = input(
+                "Pick a playerpile from 1 to 5, if you see this again, it means you picked a playerpile, with a color that you don't have, pick another one"
+            )
```

Comment:
Elég lenne azt bekérni, hogy melyik kártyáját akarja kijátszani, hiszen annak a színe azt is meghatározza, hogy hova játssza ki.

Ráadásul most azelőtt kell döntenie, hogy látta a kártyáit.

Meg látok benne egy olyan hibalaehetőséget, hogy olyan színt választ, amiből van kártyája, de mind kisebb a korábban kijáttszottnál, így egyiket sem tudja lerakni, visszalépni meg nem lehet.